### ISSUE TITLE
`terraform import` claims to succeed, but doesn't import anything, when used on a resource with `for_each`, and an incorrect index 

### ISSUE THREAD

**Issue by maxb** [hashicorp/terraform#33292]
### Terraform Version

```shell
Reproduced with both 1.4.6 and 1.5.0-beta2
```

### Terraform Configuration Files

To reproduce, you just need any resource that supports being imported. I chose `vault_auth_backend` because I am very familiar with Vault, but it should be possible to substitute any other resource with import support.

```terraform
resource "vault_auth_backend" "this" {
  for_each = {
    "wibble" = "userpass"
  }

  path = each.key
  type = each.value
}
```

### Debug Output

Not relevant here.

### Expected Behavior

Attempting an import with an inappropriate or missing `for_each` index, should generate the error:
```
Error: resource address "<address>" does not exist in the configuration.
```
in the same way importing a completely incorrect resource address does.

### Actual Behavior

```
$ terraform import 'vault_auth_backend.this["an incorrect for_each key that is not in the config"]' wibble 

Import successful!

The resources that were imported are shown above. These resources are now in
your Terraform state and will henceforth be managed by Terraform.

```

(Terraform is lying about this being a successful operation.)

### Steps to Reproduce

1. Create any simple Terraform configuration containing a single resource, that is importable, and uses for_each.
2. Try to import it, specifying a string inside the square brackets, that is not part of the `for_each` map defined in the configuration.
3. Also, try to import it, specifying a number inside the square brackets, as would be correct for resources using `count`.
4. Also, try to import it, omitting the square brackets from the resource address entirely, as would be correct for resources using neither `count` nor `for_each`.

All of these imports erroneously claim to succeed.

### References

Evidence of real-world user confused by lack of accurate diagnostics: https://discuss.hashicorp.com/t/terraform-plan-after-import-showing-add-destroy-for-an-existing-resource/54395

**Comment by maxb** [comment]
This issue I have opened is somewhat similar to #32329, but focusses on the fact that Terraform claims a success despite not doing anything, whereas the other issue focusses on Terraform having changed behaviour from earlier versions which would perform the input anyway.

**Comment by jbardin** [comment]
Thanks @maxb,

I'm tracking all versions if the report via #32329. It's all the same underlying bug, so we can close it via a single issue.

Thanks!

**Comment by maxb** [comment]
Hmm... It seems to me that just fixing the incorrect feedback to the user, would be an immediate positive move, that wouldn't need to wait on a solution to the more complex underlying problem?

**Comment by jbardin** [comment]
The lack of an error is the only underlying problem to be addressed in that issue. Importing requires configuration, so the lack of a resource instance in the configuration should raise an error.

Any improvements to the import mechanism are going to be handled via the new `import` block, and even then it's likely that importing to a non-existent expanded resource instance will remain an error as it's not something that we can really generate configuration for.
